packages/jellyfish-transaction/src/script/dftx/dftx_icxorderbook.ts

```diff
@@ -73,3 +73,35 @@ export class CICXSubmitDFCHTLC extends ComposableBuffer<ICXSubmitDFCHTLC> {
     ]
   }
 }
+
+/**
+ * ICXSubmitEXTHTLC DeFi transaction
+ */
+export interface ICXSubmitEXTHTLC {
+  offerTx: string // -----------| 32 byte, txid for which offer is this HTLC
+  amount: BigNumber // ---------| 8 byte, amount that is put in HTLC
+  hash: string // --------------| 32 byte, hash for the hash lock part
+  htlcScriptAddress: string // -| n = VarUInt{1-9 bytes}, + n bytes, script address of external htlc
+  ownerPubkey: string // -------| n = VarUInt{1-9 bytes}, + n bytes, pubkey of the owner to which the funds are refunded if HTLC timeouts
+  timeout: number // -----------| 4 byte, timeout (absolute in block) for expiration of external htlc in external chain blocks
+}
+
+/**
+ * Composable ICXSubmitEXTHTLC, C stands for Composable.
+ * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+ */
+export class CICXSubmitEXTHTLC extends ComposableBuffer<ICXSubmitEXTHTLC> {
+  static OP_CODE = 0x34 // '4'
+  static OP_NAME = 'OP_DEFI_TX_ICX_SUBMIT_EXT_HTLC'
+
+  composers (msg: ICXSubmitEXTHTLC): BufferComposer[] {
+    return [
+      ComposableBuffer.hexBEBufferLE(32, () => msg.offerTx, v => msg.offerTx = v),
+      ComposableBuffer.satoshiAsBigNumber(() => msg.amount, v => msg.amount = v),
+      ComposableBuffer.hexBEBufferLE(32, () => msg.hash, v => msg.hash = v),
+      ComposableBuffer.varUIntUtf8BE(() => msg.htlcScriptAddress, v => msg.htlcScriptAddress = v),
```

Comment:
oh this is interesting, I was expecting a script

Reply:
Had a quick look on this. `htlcScriptAddress` is the BTC HTLC address which is in base58check format(as i can remember). I think this is only for verification purposes. The claimer has to create the same BTC HTLC on his node once the data is submitted by the other party via `icx_submitexthtlc`. When the claimer creates the htlc, then the network prefixes are taken into account based on the network the node is connected to. which is the answer to your last question I believe. refer this file https://github.com/DeFiCh/ain/blob/ff53dcee23db2ffe0da9b147a0a53956f4e7ee31/src/spv/bitcoin/BRChainParams.cpp#L123